src/WebJobs.Extensions.DurableTask/DurableOrchestrationClient.cs

```diff
@@ -106,21 +106,19 @@ public override async Task<string> StartNewAsync(string orchestratorFunctionName
 
         /// <inheritdoc />
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1030:UseEventsWhereAppropriate", Justification = "This method does not work with the .NET Framework event model.")]
-        public override async Task RaiseEventAsync(string instanceId, string eventName, object eventData)
+        public override Task RaiseEventAsync(string instanceId, string eventName, object eventData)
```

Comment:
Why are we changing this from async? Also, how does this commit fix the infinite loop bug / what was it? (Was it ever filed as an issue?)

Reply:
Regarding `async`, it's more efficient and is easier on the debugging experience to have a method return a task directly rather than awaiting a task.  I'm pretty consistent about coding this way.

Regarding the infinite loop: this was caused by a PR I recently accepted, so it was limited to the dev branch. It went uncaught because I forgot to tag the PR will "full-ci" (I'm not sure how the PR author didn't notice it when testing locally) and assumed all tests ran when I saw the green check-mark in GitHub. I wonder whether there is some process we can look into which ensures that tests are actually run before PRs are accepted, or at least makes it more clear that a full CI has not yet been run.